src/Dev/Deprecation.php

```diff
@@ -163,8 +183,59 @@ protected static function get_called_method_from_trace($backtrace, $level = 1)
         if ($level == 4 && ($backtrace[$newLevel]['class'] ?? '') === InjectionCreator::class) {
             $newLevel = $newLevel + 4;
         }
+        // handle noticeWithNoReplacment()
+        foreach ($backtrace as $trace) {
+            if (($trace['class'] ?? '') === Deprecation::class
+                && ($trace['function'] ?? '') === 'noticeWithNoReplacment'
+            ) {
+                $newLevel = $newLevel + 1;
+                break;
+            }
+        }
         $called = $backtrace[$newLevel] ?? [];
-        return ($called['class'] ?? '') . ($called['type'] ?? '') . ($called['function'] ?? '');
+        return $called;
+    }
+
+    private static function isCalledFromSupportedCode(array $backtrace): bool
+    {
+        $called = Deprecation::get_called_from_trace($backtrace, 1);
+        $file = $called['file'] ?? '';
+        if (!$file) {
+            return false;
+        }
+        return Deprecation::fileIsInSupportedModule($file);
+    }
+
+    /**
+     * Check whether a file (path to file) is in a supported module
+     */
+    public static function fileIsInSupportedModule(string $file): bool
+    {
+        // Cache the supported modules list
+        if (count(Deprecation::$supportedModuleDirectories) === 0) {
+            // Manually load the supported modules list rather than use MetaData::getAllRepositoryMetaData()
+            // because we do not want to make a network request which could slow down a website
+            // While there is a small risk of the list being out of date, there is minimal downside to this
+            $path = Path::join(BASE_PATH, 'vendor/silverstripe/supported-modules/repositories.json');
+            if (!file_exists($path)) {
+                throw new RuntimeException('Could not find supported modules list');
+            }
+            $json = json_decode(file_get_contents($path), true);
+            if (is_null($json)) {
+                throw new RuntimeException('Could not parse supported modules list');
+            }
+            $dirs = array_map(fn($module) => "/vendor/{$module['packagist']}/", $json['supportedModules']);
+            // This is a special case for silverstripe-framework when running in CI
+            // Needed because module is run in the root folder rather than in the vendor folder
+            $dirs[] = '/silverstripe-framework/';
+            Deprecation::$supportedModuleDirectories = $dirs;
+        }
+        foreach (Deprecation::$supportedModuleDirectories as $dir) {
+            if (str_contains($file, $dir)) {
+                return true;
+            }
```

Comment:
Might be better to process `$file` down so it goes from `/www/html/vendor/silverstripe/framework/_config/config.yml` (or whatever it currently looks like) and get the `vendor/silverstripe/framework` part - then you can just do a `in_array()` check.

Bare in mind that while we don't have toooo many modules to loop through, this is happening for everything the config module throws for it, which at best is every config file but looks at a glance like it might be every yaml file it finds in general.

Reply:
I wouldn't worry about it :-)

```php
$n = 100000;
$start = microtime();
$str = 'the-quick-brown-fox/something-something/blah-blah';
for ($i = 0; $i < $n; $i++) {
    str_contains($str, 'fox/something');
}
$end = microtime();
$diff = $end - $start;
echo $diff . "\n";
// 0.038305
```